impy/models/sophia.py

```diff
@@ -5,158 +5,191 @@
 '''
 
 import numpy as np
-from impy.common import standard_particles
-
-
-class SophiaCascadeRun():
-    def __init__(self,
-                 lib_str,
-                 label,
-                 decay_mode,
-                 n_events,
-                 fill_subset=False,
-                 p_debug=False,
-                 nucleon_Ekin=None):
-        from ParticleDataTool import SibyllParticleTable
-        exec("import " + lib_str + " as siblib")
-        self.lib = siblib  # @UndefinedVariable
-        self.label = label
-        self.nEvents = n_events
-        self.fill_subset = fill_subset
-        self.spectrum_hists = []
-        self.dbg = p_debug
-        self.nucleon_Ekin = nucleon_Ekin
-        set_stable(self.lib, decay_mode, self.dbg)
-        self.ptab = SibyllParticleTable()
-        self.init_generator()
-
-    def init_generator(self):
+from impy.common import MCRun, MCEvent, impy_config
+from impy.util import info
+
+
+
+class SophiaEvent(MCEvent):
+    """Wrapper class around Sophia code"""
+
+    def __init__(self, lib, event_kinematics, event_frame):   
+
+        lib.toevt() # prepare hepevt common block     
+        np = lib.hepevt.nhep # number of particles in event
+        pem_arr = lib.s_plist.p[:np,:].T # array of particle momenta
+        vt_arr = lib.hepevt.vhep[:, 0:np] # array of verticies (x, y, z, t) - all zeros for sophia
```

Comment:
Comment seems to be over the line length limit

Reply:
Fixed

Reply:
The comment is moved to the line above